### DISCUSSION TITLE
Documentação e outras questões

### DISCUSSION THREAD

**Discussion by DiogoParreira** [AlertaDengue/PySUS#219] [Q&A]
Olá 

Tenho algumas questões que gostava que me conseguissem ajudar. As minhas questões são todas relacionadas com o SINAN. É possível que existam páginas a explicar algumas das questões que tenho mas procurei e não consegui encontrar. Se for o caso, por favor não hesite em redirecionar-me para o local onde posso obter esta informação. Obrigado desde já!

1) Estou a trabalhar com doenças transmitidas por vectores e queria aceder ao máximo de informação sobre casos das mesmas. Noto que na base de dados existem várias doenças. Queria saber se os meta dados que recolhem sobre os pacientes é igual para todas as doenças ? Noto por exemplo que no caso do Dengue, existem registos desde 2000 até 2024. No entanto, vejo que algumas das variáveis mudaram de nome ou deixaram de ser preenchidas com o passar do tempo e novas emergiram. Será que existe algum documento que explique as alterações à informação que foi recolhida/disponibilizada na base de dados. Noto que deixaram de recolher a variável DENGUE em 2007, Dt Nascimento so aparece em 2021. Infelizmente a página com a documentação das variáveis está em baixo 
 - https://pysus.readthedocs.io/pt/latest/tutorials/www.sgc.goias.gov.br/upload/arquivos/2013-08/dengue2.pdf

2) Cada coluna dos diferentes datasets corresponde a um caso reportado, segundo a vossa documentação. No entanto, quando pesquiso na internet vejo que muitas vezes se falam de "casos prováveis". Será que me consegue dizer se os casos nas bases de dados são todos confirmados? Se forem prováveis, será que me consegue dizer como obtenho informação sobre como foi decidido reportar e como diferenciar entre os dois? 

3) Noto que nos primeiros anos da recolha (2000 até 2006) nos casos de Dengue, não é possível converter a NU_IDADE_N para anos de vida. Até porque a variável NU_IDADE_N tem outro formato. Será que existe forma de converter esta coluna para uma int com a idade do paciente usando outra função/formula? 

4) Correndo o mesmo código para baixar os dados, obtenho diferentes números de rows para o mesmo ano. Por ex ontem baixei todos os anos de Dengue e obtive 270000 casos de Dengue para 2021. O que me pareceu um número demasiado redondo. Hoje, correndo o mesmo código, otenho 488590. Será que me conseguiam disponibilizar o número correcto de casos por ano, para fazer cross check e garantir que estou a obter todos os casos ?

Desde já muito obrigado!

**Comment by fccoelho** [comment]
Oi Diogo,

vou tentar responder às suas perguntas, são muitos assuntos e muitas das respostas podem ser melhor respondidas pela documentação oficial do SINAN
1. O SINAN é uma Base de dados que Agravos de notificação obrigatória, portanto não contém notificações de todas as doenças transmitidas por vetores. As informações disponíveis para cada agravo presente no SINAN, correspondem às informações coletadas na ficha de notificação, que **não** são iguais para todos os agravos. Além disso, o dado que o ministério publiciza, e o que o PySUS facilita o acesso, é apenas uma parte dos dados contidos na ficha de notificação. Os critérios para definição de quais variáveis são publicadas, não está documentado, até onde eu saiba. E pode mudar com o tempo.
2. Os casos nas tabelas publicadas do SINAN são **casos notificados**, casos prováveis são os casos notificados menos os casos descartados. O descarte de casos ocorre na maioria das vezes após a notificação, e neste caso as tabelas (já publicadas) são modificadas para refletir este descarte. Logo, os anos mais antigos contém casos prováveis apenas. Como as tabelas são atualizadas semanalmente, as mais recentes (digamos até 10 semanas de publicadas) ainda precisam ser tratadas como casos notificados. @sarasouzabh, você teria algo mais a acrescentar aqui?
3. Acho que o algoritmo de codificação da Idade pode ter sido alterado, mas não tenho certeza, veja o código de conversão existente no PySUS, e caso não esteja convertendo corretamente abra um "bug report" por favor.
4. Não temos como atendê-lo nesta verificação, por não mantermos registros locais das tabelas do DATASUS. você terá que verificar se todos dados foram baixados  na primeira tentativa. às vezes ocorrem time-outs silenciosos no servidor do DATASUS.

Boa sorte com as suas pesquisas.
